CRM/Core/Payment/eWAY.php

```diff
@@ -341,59 +336,21 @@ public function doDirectPayment(&$params) {
     if (self::isError($eWAYResponse)) {
       $eWayTrxnError = $eWAYResponse->Error();
       CRM_Core_Error::debug_var('eWay Error', $eWayTrxnError, TRUE, TRUE);
-      if (substr($eWayTrxnError, 0, 6) == "Error:") {
-        return self::errorExit(9008, $eWayTrxnError);
+      if (substr($eWayTrxnError, 0, 6) === 'Error:') {
+        throw new PaymentProcessorException($eWayTrxnError, 9008);
       }
       $eWayErrorCode = substr($eWayTrxnError, 0, 2);
       $eWayErrorDesc = substr($eWayTrxnError, 3);
 
-      return self::errorExit(9008, "Error: [" . $eWayErrorCode . "] - " . $eWayErrorDesc . ".");
-    }
-
-    //-----------------------------------------------------------------------------------------------------
-    // Cross-Check - the unique 'TrxnReference' we sent out should match the just received 'TrxnReference'
-    //
-    // PLEASE NOTE: If this occurs (which is highly unlikely) its a serious error as it would mean we have
-    //              received an OK status from eWAY, but their Gateway has not returned the correct unique
-    //              token - ie something is broken, BUT money has been taken from the client's account,
-    //              so we can't very well error-out as CiviCRM will then not process the registration.
-    //              There is an error message commented out here but my preferred response to this unlikley
-    //              possibility is to email 'support@eWAY.com.au'
-    //-----------------------------------------------------------------------------------------------------
-    $eWayTrxnReference_OUT = $eWAYRequest->GetTransactionNumber();
-    $eWayTrxnReference_IN = $eWAYResponse->InvoiceReference();
-
-    if ($eWayTrxnReference_IN != $eWayTrxnReference_OUT) {
-      // return self::errorExit( 9009, "Error: Unique Trxn code was not returned by eWAY Gateway. This is extremely unusual! Please contact the administrator of this site immediately with details of this transaction.");
```

Comment:
As this is commented out it can fo & then the previous lines are not used & the 'good-intentions' comment block is kinda done for too